syntax = "proto3";

package t_catalogue_proto;

import "graph.proto";

message EdgeInfo {
	uint64 edge_id = 1;
	bytes stop_name = 2;
	string bus_name = 3;
	int32 count = 4;
}

message RouteInternalData {
	oneof TEST {
		uint32 prev_edge = 2;
	}
	double weight = 3;
		
}

message HelpOpt {
	oneof h_opt{
		RouteInternalData help_opt = 1;
	}
}

message HelpRepeated {
	repeated HelpOpt help_repeated = 1;
}

message RoutesInternalData {
	repeated HelpRepeated routes = 1;
}

message Router {
	RoutesInternalData routes_internal_data = 1;
}

message TransportRouter {
	Graph graph = 1;
	repeated EdgeInfo edges_ = 2;
	Router router_ = 3;
	int32 wait = 4;
	double speed = 5;
}